src/include/network/lan/address.rb

```diff
@@ -1117,82 +1073,87 @@ def AddressDialog
 
         # When virtual interfaces are added the list of routing devices needs
         # to be updated to offer them
-        LanItems.add_current_device_to_routing if LanItems.update_routing_devices?
+        # FIXME: not working in network-ng - but should not be needed at all
+        # it updates list of available interfaces which is done elsewhere already
+        # LanItems.add_current_device_to_routing if LanItems.update_routing_devices?
       end
 
-      if LanItems.type == "vlan"
-        LanItems.vlan_etherdevice = Ops.get_string(@settings, "ETHERDEVICE", "")
-        LanItems.vlan_id = Builtins.tostring(
-          Ops.get_integer(@settings, "VLAN_ID", 0)
+      if builder.type == "vlan"
+        builder.set(option: "ETHERDEVICE", value: Ops.get_string(@settings, "ETHERDEVICE", ""))
```

Comment:
do we plan to keep this `@settings` if so, then I would prefer something like `builder.load_settings(@settings)` which will do this transformation inside.

Reply:
I currently don't see any benefit in keeping it.